Dockerfile.try

```diff
@@ -0,0 +1,133 @@
+        build-essential \
+        clang \
+        clang-11 \
+        gcc \
+        git \
+        jq \
+        libssl-dev \
+        llvm-11 \
+        make \
+        postgresql-server-dev-15 \
+        pkg-config && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set up permissions so that the postgres user can install the plrust plugin
+RUN chmod a+rwx `$(which pg_config) --pkglibdir` `$(which pg_config) --sharedir`/extension
+
+USER postgres
+ENV USER postgres
+
+# Install Rust
+RUN wget -qO- https://sh.rustup.rs | sh -s -- -y --profile minimal --default-toolchain=1.67.1
+ENV PATH="/var/lib/postgresql/.cargo/bin:${PATH}"
+
+# Copy in plrust source
+COPY --chown=${USER} . src/
+WORKDIR /src
+
+# # Install exact cargo-pgx that is definied in Cargo.toml
+# RUN TARGET_VERSION=$(cargo metadata --format-version 1 | jq -r '.packages[]|select(.name=="pgx")|.version') && cargo install cargo-pgx --force --version "$TARGET_VERSION"
+
+# # Install llvm-tools-preview
+# RUN rustup component add llvm-tools-preview rustc-dev
+
+# # Build and install plrustc
+# RUN cd plrustc && ./build.sh && cp ../build/bin/plrustc ~/.cargo/bin && find . -type d -name target | xargs rm -r
+
+# # Init cargo pgx against installed postgres version
+# RUN cargo pgx init --pg15 $(which pg_config)
+
+# # Build and install plrust as trusted
+# RUN cd plrust && STD_TARGETS="$(uname -m)-postgres-linux-gnu" ./build && cargo pgx install --release --features trusted && find . -type d -name target | xargs rm -r
+
+RUN TARGET_VERSION=$(cargo metadata --format-version 1 | jq -r '.packages[]|select(.name=="pgx")|.version') && \
+    cargo install cargo-pgx --force --version "$TARGET_VERSION" && \
+    rustup component add llvm-tools-preview rustc-dev && \
+    cd /src/plrustc && ./build.sh && cp ../build/bin/plrustc ~/.cargo/bin && \
+    cargo pgx init --pg15 $(which pg_config) && \
+    cd /src/plrust && STD_TARGETS="$(uname -m)-postgres-linux-gnu" ./build && \
+    cargo pgx install --release --features trusted && \
+    cd /src && find . -type d -name target | xargs rm -r && \
+    rustup component remove llvm-tools-preview rustc-dev
+
+# Reset the permissions of files/directories that was created or touched by the postgres user
+# Switching to the root user here temporarily is easier than installing and setting up sudo
+USER root
+
+RUN chmod 0755 `$(which pg_config) --pkglibdir` && \
+    cd `$(which pg_config) --pkglibdir` && \
+    chown root:root *.so && \
+    chmod 0644 *.so && \
+    chmod 755 `$(which pg_config) --sharedir`/extension && \
+    cd `$(which pg_config) --sharedir`/extension && \
+    chown root:root *
+
+USER postgres
+
+# Create initial main database
+RUN pg_createcluster 15 main
+
+# Set up plrust configuration options
+RUN cat <<EOT >> /etc/postgresql/15/main/postgresql.conf
+shared_preload_libraries='plrust'
+plrust.work_dir='/tmp'
+EOT
+
+# Set up custom prompt
+RUN cat <<EOT >> ~/.psqlrc
+\set PROMPT1 '%/(plrust)=# '
+\set PROMPT2 '%/(plrust)-# '
+EOT
```

Comment:
Probably a dumb question, but what does the "T" stand for?  I always write heredocs with EOF where "F" means "file".  Never seen "EOT"

Reply:
In reality it doesn't matter, the heredoc delimiter could be specified as `FOOBAR` and it wouldn't make a difference. I think I copied the structure from somewhere else.

If it makes it more readable I can change it to EOF.